iree/tools/iree-benchmark-module-main.cc

```diff
@@ -58,76 +59,15 @@ ABSL_FLAG(std::string, function_inputs_file, "",
-  return iree::OkStatus();
-}
-
-// Creates VM function instance with its inputs.
-Status PrepareIREEVMFunction(
-    iree_vm_instance_t** instance, iree_hal_device_t** device,
-    iree_vm_module_t** hal_module, iree_vm_context_t** context,
-    iree_vm_module_t** input_module, iree_vm_function_t& function,
-    iree::vm::ref<iree_vm_list_t>& inputs,
-    std::vector<iree::RawSignatureParser::Description>& output_descs,
-    const std::string& function_name, const std::string& module_data) {
-  IREE_RETURN_IF_ERROR(iree_hal_module_register_types());
-  IREE_RETURN_IF_ERROR(
-      iree_vm_instance_create(iree_allocator_system(), instance));
-
-  // Create IREE's device and module.
-  IREE_RETURN_IF_ERROR(iree::CreateDevice(absl::GetFlag(FLAGS_driver), device));
-  IREE_RETURN_IF_ERROR(CreateHalModule(*device, hal_module));
-  IREE_RETURN_IF_ERROR(LoadBytecodeModule(module_data, input_module));
-
-  // Order matters. The input module will likely be dependent on the hal
-  // module.
-  std::array<iree_vm_module_t*, 2> modules = {*hal_module, *input_module};
-  IREE_RETURN_IF_ERROR(iree_vm_context_create_with_modules(
-      *instance, modules.data(), modules.size(), iree_allocator_system(),
-      context));
-
-  IREE_RETURN_IF_ERROR(
-      (*input_module)
-          ->lookup_function(
-              (*input_module)->self, IREE_VM_FUNCTION_LINKAGE_EXPORT,
-              iree_string_view_t{function_name.data(), function_name.size()},
-              &function));
-  IREE_RETURN_IF_ERROR(ValidateFunctionAbi(function));
-
-  // Construct inputs.
-  IREE_ASSIGN_OR_RETURN(auto input_descs, ParseInputSignature(function));
-
-  if (!absl::GetFlag(FLAGS_function_inputs_file).empty()) {
-    IREE_ASSIGN_OR_RETURN(inputs,
-                          ParseToVariantListFromFile(
-                              input_descs, iree_hal_device_allocator(*device),
-                              absl::GetFlag(FLAGS_function_inputs_file)));
-  } else {
-    IREE_ASSIGN_OR_RETURN(
-        inputs,
-        ParseToVariantList(input_descs, iree_hal_device_allocator(*device),
-                           absl::GetFlag(FLAGS_function_inputs)));
-  }
-
-  // Creates output singnature.
-  IREE_ASSIGN_OR_RETURN(output_descs, ParseOutputSignature(function));
-  return iree::OkStatus();
-}
-
 void RegisterModuleBenchmarks(
-    Status module_data_status, Status prepare_vm_func_status,
-    std::string& function_name, iree_vm_context_t* context,
+    const std::string& function_name, iree_vm_context_t* context,
     iree_vm_function_t function, iree_vm_list_t* inputs,
     const std::vector<RawSignatureParser::Description>& output_descs) {
   auto benchmark_name = "BM_" + function_name;
   benchmark::RegisterBenchmark(
       benchmark_name.c_str(),
-      [module_data_status, prepare_vm_func_status, context, function, inputs,
+      [context, function, inputs,
        output_descs](benchmark::State& state) -> void {
```

Comment:
Nit: formatting looks funky here. What happens if you stick it on one line and then ask clang-format to format it. Is it really this?

Reply:
Yes, I was also curious about this, and it is what it is.